src/main/java/org/mule/tools/devkit/sonar/checks/maven/DevkitLatestVersionCheck.java

```diff
@@ -0,0 +1,121 @@
+        }
+        return issues;
+    }
+
+    public Version getLatestVersion() throws IOException, XMLStreamException {
+        XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+        InputStream in = new URL("https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/tools/devkit/mule-devkit-parent/maven-metadata.xml").openStream();
+        XMLStreamReader streamReader = inputFactory.createXMLStreamReader(in);
+        Version latestVersion;
+        Version currentVersion;
+        String currentTag = "";
+        String currentValue;
+        streamReader.nextTag(); // Advance to "metadata" element
+        do{
+            streamReader.next();
+            if(streamReader.hasName()){
+                currentTag = streamReader.getName().toString();
+            }
+        }while(!currentTag.equals("versions") && streamReader.hasNext()); // Advance until we reach versions
+        streamReader.nextTag(); // Advance into the first element inside versions
+        streamReader.next(); // Advance into the value of the first version
+        currentValue = streamReader.getText();
+        latestVersion = new Version(currentValue);
+        do{
+            if(streamReader.isStartElement()){
+                currentValue = streamReader.getElementText();
+            }
+            if(currentValue.indexOf('-') < 0 && !currentValue.isEmpty() && currentValue.indexOf('4') != 0){
+                currentVersion = new Version(currentValue);
+                currentVersion.replaceIfGreaterThan(latestVersion);
+            }
+            streamReader.next();
+            if(streamReader.hasName()){
+                currentTag = streamReader.getName().toString();
+            }
+            currentValue = "";
+        }while(!currentTag.equals("versions") && streamReader.hasNext());
+        in.close();
+        return latestVersion;
+    }
+
+    class Version implements Comparable<Version> {
+        int major;
+        int minor;
+        int rev;
+
+        public int compareTo(Version latestVersion) {
+            if (this.major != latestVersion.major) {
+                return Integer.compare(this.major, latestVersion.major);
+            }
+            if (this.minor != latestVersion.minor) {
+                return Integer.compare(this.minor, latestVersion.minor);
+            }
+            if (this.rev != latestVersion.rev) {
+                return Integer.compare(this.rev, latestVersion.rev);
+            }
+            return 0;
+        }
+
+        public void replaceIfGreaterThan(Version version){
+            if(this.minor > version.minor || this.minor == version.minor && this.rev > version.rev){
+                version.minor = this.minor;
+                version.rev = this.rev;
+            }
+        }
+
+        public Version (String version) {
+            List<String> tokens = Splitter.on(".").omitEmptyStrings().splitToList(version);
```

Comment:
Splitter does not handle dash ("-") if versions contain revisions. Use `Splitter.onPattern("[.-]")...`

Additionally, conversion will fail for those revision when calling ` Integer.parseInt(...);